blaze/expr/collections.py

```diff
@@ -123,8 +123,10 @@ def sort(child, key=None, ascending=True):
     if key is None and isrecord(child.dshape.measure):
         key = child.dshape.measure.names
     if isinstance(key, (list, tuple)):
-        key = keys_to_validate = tuple(key)
```

Comment:
if the key is an instance of field then we should verify that the child of the field is compatible with the eexpression being sorted.

We should not allow things like:
```python
a = bz.data(...)
b = bz.data(...)
a.sort(b.field)
```